private/buf/cmd/buf/command/build/build.go

```diff
@@ -139,6 +140,21 @@ func run(
 	if err != nil {
 		return err
 	}
+	imageRef, err := buffetch.NewImageRefParser(container.Logger()).GetImageRef(ctx, flags.Output)
+	if err != nil {
+		return fmt.Errorf("--%s: %v", outputFlagName, err)
+	}
+	var buildOptions []bufimagebuild.BuildOption
+	if flags.ExcludeSourceInfo {
+		buildOptions = append(buildOptions, bufimagebuild.WithExcludeSourceCodeInfo())
+	} else if !flags.AsFileDescriptorSet {
+		// if not a descriptor set, not necessary to try to match protoc, so include extra comments
```

Comment:
I'm not sure we want to lock ourselves to the SourceCodeInfos that protoc produces - there's no exact definition as to what SourceCodeInfos a FileDescriptorSet will produce, and us producing extra is arguably better than what protoc does. I would not change the SourceCodeInfos dependent on the state of `--as-file-descriptor-set`, instead I'd remove the `WithExtraSourceCodeInfo` option altogether and make having extra source code info the default.

Reply:
> there's no exact definition as to what SourceCodeInfos a FileDescriptorSet will produce

It is documented in `descriptor.proto`. And `protoparse` wasn't really conforming to it (intentionally -- I originally added the extra comments before I implemented the AST, because I wanted a way to get access to as much of the original source as possible).

I'm fine removing it. But it seemed valuable for maximizing trust that a descriptor set produced by buf matches that of protoc.

Reply:
Let's discuss tomorrow or in person this week